pilot/pkg/model/config.go

```diff
@@ -94,6 +95,10 @@ type Config struct {
 
 	// Spec holds the configuration object as a protobuf message
 	Spec proto.Message
+
+	// Mutex to protect this config against simultaneous reads and writes. Use
+	// it within critical blocks and call _Lock()_ only when a write is planned.
+	sync.RWMutex
```

Comment:
shouldn't this be a pointer?

Reply:
@hzxuzhonghu Having this as a pointer will get a zero value of nil and you need to init it somewhere.
As it is now will have a zero value of an unlocked mutex ready to be used. Why do you think it should be a pointer?

Reply:
Because Mutex does not support value copy. And We have value copy of Config in `List` https://github.com/istio/istio/pull/8423/files#diff-ee39cd332931ab6f8c10713ce3446d09R343

Reply:
>Values containing the types defined in this package should not be copied. 

golang doc:  https://golang.org/pkg/sync/

Reply:
I see what you meant now. The tricky part would be when and where to init it.